index.js

```diff
@@ -179,7 +179,7 @@ Request.prototype.run = function () {
                 }
             }
 
-            if (res.status >= 400) {
+            if (!res.status || res.status < 100 || res.status >= 400) {
```

Comment:
The 504 is already something we invented, to mark request timeouts with a status code.

Something in the 5xx range seems to make sense, but 500 would mix this with actual 500s returned by the service, which would be confusing. I don't see any official 5xx response code for invalid responses, so we'd probably have to invent one.